[bug]: installing @uploadthing/react for next@14.0.3 gives installation error

### Provide environment information

```bash
System:
    OS: Linux 5.15 Ubuntu 20.04.5 LTS (Focal Fossa)
    CPU: (16) x64 AMD Ryzen 9 5980HS with Radeon Graphics
    Memory: 11.37 GB / 15.33 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 20.5.1 - ~/.nvm/versions/node/v20.5.1/bin/node
    Yarn: 1.22.19 - /mnt/c/Users/woute/AppData/Roaming/npm/yarn
    npm: 9.8.0 - ~/.nvm/versions/node/v20.5.1/bin/npm
  npmPackages:
    @uploadthing/react: ^6.0.1 => 6.0.1 
    typescript: ^5 => 5.2.2 
    uploadthing: ^6.0.3 => 6.0.3
```


### Describe the bug

When installing @uploadthing/react in a next@14.0.3, it throws an error:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @uploadthing/react@6.0.1
npm ERR! Found: next@14.0.3
npm ERR! node_modules/next
npm ERR!   peer next@"^12.2.5 || ^13 || ^14" from @kinde-oss/kinde-auth-nextjs@2.0.9
npm ERR!   node_modules/@kinde-oss/kinde-auth-nextjs
npm ERR!     @kinde-oss/kinde-auth-nextjs@"^2.0.9" from the root project
npm ERR!   peer next@"*" from @trpc/next@10.43.6
npm ERR!   node_modules/@trpc/next
npm ERR!     @trpc/next@"^10.43.6" from the root project
npm ERR!   1 more (the root project)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional next@"14.0.1" from @uploadthing/react@6.0.1
npm ERR! node_modules/@uploadthing/react
npm ERR!   @uploadthing/react@"*" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: next@14.0.1
npm ERR! node_modules/next
npm ERR!   peerOptional next@"14.0.1" from @uploadthing/react@6.0.1
npm ERR!   node_modules/@uploadthing/react
npm ERR!     @uploadthing/react@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

peerDependencies need to be updated?

![image](https://github.com/pingdotgg/uploadthing/assets/76952619/15c1fcd0-557a-4c01-bdcf-2e7899b38174)


### Link to reproduction

https://github.com/pingdotgg/uploadthing/blob/2a548a8fe7e51f09f0b22af8e4e11de1c758374a/packages/react/package.json

### To reproduce

Run `npm i @uploadthing/react` inside a project that runs on `"next": "14.0.3",`

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/.changeset/warm-balloons-decide.md
+++ b/.changeset/warm-balloons-decide.md
@@ -1,0 +1,5 @@
+---
+"@uploadthing/react": patch
+---
+
+fix: @uploadthing/react peer deps

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -50,7 +50,7 @@
     "tailwind-merge": "^1.13.2"
   },
   "peerDependencies": {
-    "next": "14.0.1",
+    "next": "*",
     "react": "^17.0.2 || ^18.0.0",
     "uploadthing": "^6.0.0"
   },

